File: 06-task-buffer-overflow/vuln_code.c
#include <stdio.h>
#include <string.h>

/*
 * Exercise 06 - Buffer Overflow
 * WORKSHOP: Secure Programming
 *
 * Audit: Read this code carefully. Can you spot the security flaw?
 */

void greet_user(void)
{
	char buffer[64];   /* Only 64 bytes allocated */

	printf("What is your name? ");
	fflush(stdout);

	fgets(buffer, sizeof(buffer), stdin);

	buffer[strcspn(buffer, "\n")] = '\0';
	printf("Hello, %s!\n", buffer);
}

int main(void)
{
	greet_user();

	return 0;
}

File: 04-task-eval-injection/vuln_code.js
'use strict';

/**
 * Exercise 04 - eval() Injection
 * WORKSHOP: Secure Programming
 *
 * Audit: Read this code carefully. Can you spot the security flaw?
 *
 * A simple CLI calculator: reads a math expression and prints the result.
 */

const readline = require('readline');
const SAFE_MATH = /^[\d\s+\-*/.()]+$/;

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
    terminal: false
});

rl.question('Enter a math expression (e.g. 2+2): ', (input) => {
    try {
        if (!SAFE_MATH.test(input.trim())) {
            console.error('Error: only numeric expressions are allowed.');
            process.exit(1);
        }

        // Only reach here if input is safe
        const result = Function('"use strict"; return (' + input + ')')();
        console.log(`Result: ${result}`);
    } catch (err) {
        console.error(`Error: ${err.message}`);
    }
    rl.close();
});
